src/messageheader.h

```diff
@@ -0,0 +1,46 @@
+#include <serialize.h>
+
+#ifndef BITCOIN_MESSAGEHEADER_H
+#define BITCOIN_MESSAGEHEADER_H
+
+#include <cstdint>
+#include <string>
+
+
+/** Message header.
+ * (4) message start.
+ * (12) command.
+ * (4) size.
+ * (4) checksum.
+ */
+class CMessageHeader
+{
+public:
+    static constexpr size_t MESSAGE_START_SIZE = 4;
+    static constexpr size_t COMMAND_SIZE = 12;
+    static constexpr size_t MESSAGE_SIZE_SIZE = 4;
+    static constexpr size_t CHECKSUM_SIZE = 4;
+    static constexpr size_t MESSAGE_SIZE_OFFSET = MESSAGE_START_SIZE + COMMAND_SIZE;
+    static constexpr size_t CHECKSUM_OFFSET = MESSAGE_SIZE_OFFSET + MESSAGE_SIZE_SIZE;
+    static constexpr size_t HEADER_SIZE = MESSAGE_START_SIZE + COMMAND_SIZE + MESSAGE_SIZE_SIZE + CHECKSUM_SIZE;
+    typedef unsigned char MessageStartChars[MESSAGE_START_SIZE];
```

Comment:
Would it make more sense to pull `MessageStartChars` alone into kernel/chainparams.h, and just add a `static_assert(sizeof(MessageStartChars) == MESSAGE_START_SIZE);` here?

Reply:
Yes, that works too and I went through such an iteration. I thought adding such logic might be a bit less desirable though than just straight up moving these simple definitions and keeping the lines that are actually changed to ~0, especially since the kernel code uses definitions to both `MessageStartChars` and its size.

Reply:
I guess I look at it more as "chainparams provides a 4 byte magic number" and "other things (p2p, wallet, block store) use those magic numbers to help stop you using a different chain than you intended to". That seems like a sensible thing for chainparams to do; but it's just that one line, all the other stuff in this header is p2p protocol things which don't seem interesting for the kernel?

Reply:
I don't mind either way, this is a simple change. To be clear, is your intention: Move `MESSAGE_START_SIZE` and the `MessageStartChars` to `kernel/chainparams.h`, then include `kernel/chainparams.h` as needed?

Reply:
I believe the two approaches work just fine together? https://github.com/theuni/bitcoin/commit/23b992e661650e87f572895ee4bad0e6e5e5196a

Reply:
What @theuni said.

Reply:
> If you take that, please use less stupid names than I did :p

Pretty sure I won't :P